### ISSUE TITLE
`no-for-loop` rule produces invalid autofix in trying to use `.entries()` to iterate a string

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#738]
In situations like this where we can tell that a variable being iterated is a string (not an array), we should not convert the loop to use the `.entries()` function which does not exist on strings.

Before:
```js
const myString = 'helloworld';
for (let i = 0; i < myString.length; i++) {
    console.log(i, myString[i])
}
```

After autofix (invalid):
```js
const myString = 'helloworld';
for (const [i, element] of myString.entries()) {
    console.log(i, element);
}
```

**Comment by snowteamer** [comment]
I don't think it is possible to reliably determine the type of the variable being iterated, except in a few easy cases.

Moreover, there are other array-like objects which usually don't have an `entries()` method either:
- plain array-like objects like `{length: 4, 0: "a", 1: "b", 2: "c", 3: "d"}`
- instances of `HTMLCollection` like `document.body.children`
- ...
So in these cases I think you have to manually apply the autofix on a case-by-case basis.

**Comment by noftaly** [comment]
Maybe the suggestion API could be used, which propose either `variable.entries()` or `[...variable].entries()` for non-array variables. Although not everyone might like the [...variable] notation because it is bulky and makes the for less readable (maybe even less readable than before the fix?).
I just ran into the same issue, and if someone finds a solution I'm willing to make a PR 😄

**Comment by bmish** [comment]
I opened a fix in #1242.
